apps/accounts/automations/base/manager.py

```diff
@@ -128,30 +127,11 @@ def host_callback(self, host, asset=None, account=None, automation=None, path_di
 
         return inventory_hosts
 
-    def wait_and_save_recorder(self, recorder, max_retries=10, retry_interval=2):
-        recorder_model = type(recorder)
-
-        for attempt in range(max_retries):
-            exist = recorder_model.objects.filter(
-                account_id=recorder.account_id, execution=self.execution
-            ).exists()
-
-            if exist:
-                print(f"Data inserted, updating recorder status after {attempt + 1}th query")
-                recorder.save(update_fields=['error', 'status', 'date_finished'])
-                return True
-
-            print(f"Data not ready, waiting {retry_interval} second(s) and retrying ({attempt + 1}/{max_retries})")
-            time.sleep(retry_interval)
-
-        print("\033[31m The data is still not inserted, giving up saving the recorder status.\033[0m")
-        return False
-
-    def save_record(self, recorder):
-        self.wait_and_save_recorder(recorder)
+    @staticmethod
+    def save_record(recorder):
+        recorder.save(update_fields=['error', 'status', 'date_finished'])
 
     def on_host_success(self, host, result):
-
         recorder = self.name_recorder_mapper.get(host)
         if not recorder:
             return
```

Comment:
There appears to be no major issues with the code at this point, but there may be some minor improvements that can be made in terms of naming:

* `host_callback` function name seems a bit misleading because it's not directly associated with callback functionality.
* `wait_and_save_recorder` should ideally return only when all attempts succeed or fail.

I haven't found an actual change yet, which means these changes have been made during the development or testing phase based on their previous versions. However, I recommend reviewing and possibly adjusting the functions' names and logic according to best practices.